### PR TITLE
Add ESLint rules and mild cleanup

### PR DESCRIPTION
**Pull request by pdehaan**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -12,5 +12,20 @@
     "commonjs": true,
     "node": true,
     "es6": true
+  },
+
+  "rules": {
+    "curly": 0,
```
Comment (pdehaan) [inline .eslintrc]: Rules are the same as the other repo, but I left `curly` == 0 for now

File: lib/parsers/icon.js [hunk 2]
```diff
@@ -39,37 +39,37 @@ module.exports = function($, url) {
 
 function getIcons(selector, $, url) {
   var nodes = $(selector);
-   var items = [];
+  var items = [];
```
Comment (pdehaan) [inline lib/parsers/icon.js]: This is noisy, but apparently the indentation was slightly off.

File: lib/parsers/simple.js [hunk 3]
```diff
@@ -14,21 +14,23 @@ module.exports = function($) {
...
-      var value = description.attr('content');
+      value = description.attr('content');
       if (value) result.description = value.trim();
     }
 
     // theme color
     var themeColor = $('meta[name="theme-color"]');
     if (themeColor) {
-      var value = themeColor.attr('content');
+      value = themeColor.attr('content');
```
Comment (pdehaan) [inline lib/parsers/simple.js]: Not sure why it didn't like `let` or `const` block scopings here. So I moved the declaration up one level.

File: lib/parsers/social-user.js [hunk 4]
```diff
@@ -7,20 +8,21 @@ const PATH = require('path');
...
-    "twitter.com": function(url, doc, metadata) {
+    'twitter.com': function(url, doc, metadata) {
       const twitter = {};
       const parsedUrl = URL.parse(url);
-      const dirs = parsedUrl.pathname.split("/").filter(dirComponent => dirComponent.length);
+      const dirs = parsedUrl.pathname.split('/').filter(dirComponent => dirComponent.length);
 
-      logger.debug('SocialUserParser with directories ', dirs);
+      // TODO: error  'logger' is not defined  no-undef
+      // logger.debug('SocialUserParser with directories ', dirs);
```
Comment (pdehaan) [inline lib/parsers/social-user.js]: Commented this line out since `logger` is undefined here (the declaration is commented out on L5 above).

**Comment by wilsonpage** [comment]
Some stuff landed, needs a rebase 😄

**Comment by pdehaan** [comment]
Rebased.

**Pull request merged**